app/services/induction/delete_dup_record.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class Induction::DeleteDupRecord < BaseService
+  COMPARE_ATTRIBUTES = %(induction_programme_id participant_profile_id schedule_id training_status preferred_identity_id school_transfer appropriate_body_id)
+
+  class DeleteInductionRecordRestrictionError < StandardError; end
+
+  def call
+    ActiveRecord::Base.transaction do
+      check_record_is_deletable!
+
+      update_previous_record
+      delete_induction_record
+    end
+  end
+
+private
+
+  attr_reader :induction_record
+
+  def initialize(induction_record:)
+    @induction_record = induction_record
+  end
+
+  def check_record_is_deletable!
+    raise DeleteInductionRecordRestrictionError, "Cannot delete record because it is active" if active?
```

Comment:
I think exceptions are very slow. Can we try error messages instead to reduce that 1 day completion time?

Reply:
I ran the deletion in a 10k sample records and I got no exceptions raised. Maybe there will be some edge cases that will raise some of them, but it doesn't seem it will make any difference performance wise.